src/components/tour/tour_step.tsx

```diff
@@ -131,6 +149,26 @@ export const EuiTourStep: FunctionComponent<EuiTourStepProps> = ({
     );
   }
 
+  const [willMount, setWillMount] = useState<boolean>(false);
+  const animationFrameId = useRef<number>();
+  const node = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (anchor) {
+      animationFrameId.current = window.requestAnimationFrame(() => {
+        node.current = findElement(anchor);
+        if (node) {
+          setWillMount(true);
+        }
+      });
+    }
+
+    return () => {
+      animationFrameId.current &&
+        window.cancelAnimationFrame(animationFrameId.current);
+    };
```

Comment:
Since this is returning on every `anchor` prop change, we also should be unsetting `willMount` (or `hasValidAnchor` if you like my naming suggestion) to false in case the consuming application passes in a new anchor that's invalid:

```suggestion
    return () => {
      setHasValidAnchor(false);
      animationFrameId.current &&
        window.cancelAnimationFrame(animationFrameId.current);
    };
```

Reply:
> Or alternatively you could do this up above after line 160-162

Went this route (kind of) using a ternary